### PR TITLE
ISPN-12512 Protostream with spring boot remote

### LINKED ISSUE #121: No marshaller registered for object of Java
Running your example: https://github.com/infinispan/infinispan-simple-tutorials/tree/master/spring-integration/spring-boot/remote with the latest infinispan release (12.1.3.final). Results in following Exception:

java.lang.IllegalArgumentException: No marshaller registered for object of Java type org.infinispan.tutorial.simple.spring.remote.BasqueName : BasqueName{id=6, name='Koldo'}
	at org.infinispan.protostream.impl.SerializationContextImpl.getMarshallerDelegate(SerializationContextImpl.java:512) ~[protostream-4.4.0.Final.jar:4.4.0.Final]

The error seems to occur while deserialization, since  I can see the Entries in the Server. A (quick) Debugging in SerializationContextImpl indicates that class BasqueName neither is present in marshallersByClass nor in legacyMarshallerProviders.

Using the org.infinispan.commons.marshall.JavaSerializationMarshaller works fine.

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-12512

Closes #121

### CODE REVIEW

File: spring-integration/spring-boot/remote/src/main/java/org/infinispan/tutorial/simple/spring/remote/Reader.java [hunk 1]
```diff
@@ -23,12 +25,24 @@
    // Manipulate the underlying cache to show some messages
    private RemoteCache<Integer, String> cache;
 
-   private final String XML = String.format("<infinispan><cache-container><distributed-cache name=\"" + Data.BASQUE_NAMES_CACHE + "\"></distributed-cache></cache-container></infinispan>", "cache");
+   private final String XML = String.format("<infinispan>" +
```
[the remark this answers is not in the document]
Comment (pruivo) [reply spring-integration/spring-boot/remote/src/main/java/org/infinispan/tutorial/simple/spring/remote/Reader.java:28]: can be static too :)

[the remark this answers is not in the document]
Comment (pruivo) [reply spring-integration/spring-boot/remote/src/main/java/org/infinispan/tutorial/simple/spring/remote/Reader.java:28]: and you don't need the `<infinispan>` and `<cache-container>` tags
```
   private static final String XML =
           "<distributed-cache name=\"" + Data.BASQUE_NAMES_CACHE + "\">" +
           "<encoding media-type=\"application/x-protostream\"/>" +
           "</distributed-cache>";
```

**Comment by karesti** [comment]
@pruivo I pushed the changes

**Comment by pruivo** [comment]
I'm getting  `java.lang.IllegalArgumentException: Cannot find cache named 'basque-names'` when running your code

**Comment by karesti** [comment]
@pruivo try again plz

**Comment by pruivo** [comment]
NPEs:
```
java.lang.NullPointerException: null
	at org.infinispan.tutorial.simple.spring.remote.Reader.<init>(Reader.java:36) ~[classes/:na]
```
and
```
java.lang.NullPointerException: null
	at org.infinispan.tutorial.simple.spring.remote.Reader.retrieveSize(Reader.java:44) ~[classes/:na]
```

**Comment by karesti** [comment]
@pruivo now. sorry I totally missed that when moving the cache creation to the startup config bean. the sessions one works well for you ?

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
it is working now. is there anything missing?

**Comment by karesti** [comment]
no, I think you can merge it @pruivo

**Comment by pruivo** [comment]
integrated! thanks @karesti !

**Pull request merged**